### PR TITLE
fix: imps self-destroying on clients when late-joining with multiple additive scenes loaded

### PR DESCRIPTION
**Pull request by LPLafontaineB**
### Description (*)

This PR makes sure ClientCharacterVisualization is only enabled on clients when it is spawned. This prevents Update from being called at the wrong time, causing it to self-destruct. This PR also removes an unneeded property and moves the OnDestroy code to OnNetworkDespawn to better mirror its initialization in OnNetworkSpawn.
### Issue Number(s) (*)

Fixes issue(s): [MTT-2433](https://jira.unity3d.com/browse/MTT-2433)
### Manual testing scenarios

1. Start a game as a host
2. Start the game and move to the second door to load the TransitionArea additive scene
3. As a client, join the game instance
4. See that the imps in the first room are visible
### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs
Comment (LukeStampfli) [inline Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs:123]: Why are we changing this from the clientvis Parent to the transform.Parent. From my understanding the Parent property of the client visualization is not the actual parent of the object? XML doc says:
```
        /// <summary>
        /// Returns our pseudo-Parent, the object that owns the visualization.
        /// (We don't have an actual transform parent because we're on a top-level GameObject.)
        /// </summary>
        public Transform Parent { get; private set; }
```

Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs:123]: It got changed some time ago with a refactor of the characters. Now the clientVis is a child of the actual character, so this property is no longer necessary.

Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs:123]: I believe you can retrieve that from `m_Parent.NetState.IsNpc`, right?

File: Assets/BossRoom/Scripts/Client/Game/Action/TargetActionFX.cs
Comment (fernando-cortez) [inline Assets/BossRoom/Scripts/Client/Game/Action/TargetActionFX.cs:25]: We could cache a reference to the parent's `NetworkCharacterState`, and remove parent entirely, since it seems like it's deprecated (removes two GetComponents).

File: Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs:123]: I just noticed the removal of that property was not included in my commit. Fixed it now.

File: Assets/BossRoom/Scripts/Client/Game/Character/ClientCharacterVisualization.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/Game/Character/ClientCharacterVisualization.cs]: done

File: Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs
Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/Game/Action/FXProjectileTargetedActionFX.cs:123]: Looks good to me. @LukeStampfli I'll let you resolve.

**Comment by SamuelBellomo** [comment]
Noel is looking at this bug here https://jira.unity3d.com/browse/MTT-2924
Should we put this PR on hold?

**Comment by LPLafontaineB** [comment]
> Noel is looking at this bug here https://jira.unity3d.com/browse/MTT-2924 Should we put this PR on hold?

This PR is not related to that bug (I think you have #399 in mind). This is for an issue on our side, where in the case of multiple scenes being loaded additively, when synchronizing, there are a few frames on the client where Update and FixedUpdate are called before OnNetworkSpawn happens.

**Review by SamuelBellomo — APPROVED** [review]
approving once other conversations are solved

**Review by LukeStampfli — APPROVED** [review]

**Pull request merged**
